Donation-platform/src/main/java/com/google/servlets/PersonalizationServlet.java

```diff
@@ -48,8 +53,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       e.printStackTrace();
     }
     // Iterate over the mapping and add the tag values to the tags list
+    int numberOfBlankTags = 0;
     for (Map.Entry<String, String> entry : tagMap.entrySet()) {
-      tags.add(entry.getValue());
+      String tag = entry.getValue();
+      if(tag.length() == 0) {
+        numberOfBlankTags++;
```

Comment:
why can't we directly check size of  tagMap using "size" and check the count?

Reply:
Because a tag selection field left blank results in an empty string in the tags list. For example, if only tag 2 was selected, the tags list would be like this: [ , education, ].

Reply:
Because a tag selection field left blank is translated into an empty string. If no tags are selected in the html form, requestData looks like {"tag1":"","tag2":"","tag3":""}, which, when converted into a map, is still a 3 element mapping ("size" would think there's three tags when there's actually zero).

Reply:
got it, you can directly check map.getValue().isEmpty() instead of checking length

Reply:
Done.